src/services/Claudy.jsx

```diff
@@ -205,6 +210,7 @@ export class Claudy extends Component {
                 iconSrc={this.getIcon(selectedAction.icon)}
                 url={selectedActionUrl}
                 onActionClick={() => this.trackActionLink(selectedAction)}
+                onSuccess={this.goBack}
```

Comment:
I think we should ensure that we are using the right context, even if it seems to work as it.
```js
    onSuccess={() => this.goBack()}
```

Reply:
Hmm, agree, but I need to pass an optional argument, see https://github.com/cozy/cozy-settings/pull/71/files#diff-262290da52d26b12b91bcea1395c10e6R125 and https://github.com/cozy/cozy-settings/pull/71/files#diff-dad91f6e3ef7c4e1710f2ad844a0f6e7R29.

So maybe we could just bind the function to the context: `onSuccess={this.goBack.bind(this)}`, so we keep the context and we can pass args if needed. Is it better?

Reply:
and what about `{(alert) => this.goBack(alert)}` ?

Reply:
Actually, I personnally try to avoid binding like this, I feel them confusing.
Furthermore, @goldoraf told me back in the days that the notation `foo={() => bar()}` was more "React". I'd love to have him confirming this point.

Reply:
Weird that it works this way, without binding `this`... `foo={() => bar()}` is a great way to avoid binding this at a small performance cost, another way is using [class properties](https://babeljs.io/docs/plugins/transform-class-properties/).

Reply:
Please read this little article (~ 3 minutes) about arrow functions or bind in props by Cory House : https://medium.freecodecamp.org/why-arrow-functions-and-bind-in-reacts-render-are-problematic-f1c08b060e36

Reply:
Isn't the case presented in the article only related to having an arrow function into a `map` into a `render` ?
I think that we are not concerned here, but it is still really interesting to keep in mind what is explained in the article.

Reply:
Lorsque tu passes une arrow function ou que tu bind une fonction dans la props, le composant parent P passe à chaque fois qu'il est rendu une nouvelle instance de cette props à ses enfants.
Ainsi `PureComponent` et `shouldComponentUpdate` seront cassés car ce seront des nouvelles instances, donc pas de comparaison possible pour le VDOM.

> The parent component is passing down an arrow function on props. Arrow functions are reallocated on every render (same story with using bind). So although I’ve declared User.js as a PureComponent, the arrow function in User’s parent causes the User component to see a new function being sent in on props for all users. So every user re-renders when any delete button is clicked. 👎

Si tu passes une même instance à la props des enfants, alors cette instance pourra être comparée et re-rendu seulement si nécessaire.

Regarde le deuxième exemple de l'article. Il utilise une class properties pour déclarer ses composants enfants (permettant le scoping de this) et on voit bien dans la console que les enfants ne sont pas re-rendus à chaque delete: https://codesandbox.io/s/jnowr0ww7v

Reply:
Attention : mon message n'est pas un appel à faire autrement, c'est juste pour informer de l'inconvénient de arrow/bind dans les props. Les problèmes d'optimisation viendront en temps voulu.